packages/loopring_v3/contracts/amm/libamm/AmmBlockReceiver.sol

```diff
@@ -56,37 +58,43 @@ library AmmBlockReceiver
     {
         uint size = S.tokens.length;
         return AmmData.Context({
-            _block: _block,
             txIdx: txIdx,
             exchange: S.exchange,
-            exchangeDomainSeparator: S.exchange.getDomainSeparator(),
+            exchangeDomainSeparator: S.exchangeDomainSeparator,
             domainSeparator: S.domainSeparator,
             accountID: S.accountID,
             poolTokenID: S.poolTokenID,
             totalSupply: S._totalSupply,
-            size: size,
             tokens: S.tokens,
-            tokenBalancesL2: new uint96[](size)
+            tokenBalancesL2: new uint96[](size),
+            transactionBuffer: AmmData.TransactionBuffer({
+                size: 0,
+                owners: new address[](size*2 + 1),
```

Comment:
size  + 2?
For each token, we need one transfer (+size), and we also need a pool token transfer (+1) and an AmmUpdate (+1)

Reply:
We need an AmmUpdate for each token (have to get the balance for each token separately). So I think `size * (transfer + update) + burn/mint`.

Reply:
I see.